src/commands/upload_file.ts

```diff
@@ -32,20 +35,40 @@ interface UploadFileParameter {
 	driveKey?: DriveKey;
 }
 
+type FilePath = string;
+
+const formatResults = (results: ArFSResult[]): ArFSResult => {
+	const isFolderUploadResults = results.length === 1;
+	if (isFolderUploadResults) {
+		return results[0];
+	}
+	return results.reduce(
```

Comment:
Doesn't think generalize to the results.length === 1 case? In which case the special case above goes away.

Reply:
We could have just the `reduce` as it works for both cases, but there's no need to iterate over results when you have only one result. I thought that it would be less expensive to just check for it instead of running the spread operators for nothing.

Do you think that makes the code more unreadable?

Reply:
I agree that your solution is more performant in the single item case! But I think it's trading off a very tiny bit of performance for more lines of code that need to be tested, read, and comprehended. I think the reduce is clear and performant "enough" :)